runtime/parser/expression_test.go

```diff
@@ -5901,35 +5913,45 @@ func TestParseIdentifiers(t *testing.T) {
 		"FOO_______",
 		"Fo123__21341278AAAAAAAAAAAAA",
 	}
-	for _, name := range names {
-		t.Run(name, func(t *testing.T) {
 
-			code := fmt.Sprintf(`let %s = 1`, name)
-			_, errs := testParseProgram(code)
-			require.Empty(t, errs)
-		})
-	}
+	testParseIdentifiersWith(t, names, func(t *testing.T, _ string, errs error) {
+		require.Empty(t, errs)
+	})
 }
 
-func TestParseReservedIdent(t *testing.T) {
+func TestParseHardKeywords(t *testing.T) {
 	t.Parallel()
 
-	for keyword := range hardKeywords {
-		code := fmt.Sprintf(`let %s = 0`, keyword)
-		_, err := testParseProgram(code)
-		upcast := err.(Error)
-		errs := upcast.Errors
+	// seriously why doesn't golang have iterators
+	hardKeywordList := make([]string, len(hardKeywords))
+	for kw := range hardKeywords {
+		hardKeywordList = append(hardKeywordList, kw)
+	}
 
+	testParseIdentifiersWith(t, hardKeywordList, func(t *testing.T, keyword string, err error) {
 		utils.AssertEqualWithDiff(t,
 			[]error{
 				&SyntaxError{
 					Pos:     ast.Position{Line: 1, Column: 4, Offset: 4},
 					Message: "expected identifier after start of variable declaration, got keyword " + keyword,
 				},
 			},
-			errs,
+			err.(Error).Errors,
 		)
+	})
+}
+
+func TestParseSoftKeywords(t *testing.T) {
+	t.Parallel()
+
+	softKeywordList := make([]string, len(softKeywords))
```

Comment:
Same as above, you probably just want to preallocate the capacity:

```suggestion
	softKeywordList := make([]string, 0, len(softKeywords))
```